src/autocomplete.js

```diff
@@ -403,39 +440,46 @@ class Autocomplete {
         this.getCompletionProvider({
             prefix,
             pos
-        }).provideCompletions(this.editor, completionOptions, function(err, completions, finished) {
-            var filtered = completions.filtered;
-            var prefix = util.getCompletionPrefix(this.editor);
-
-            if (finished) {
-                // No results
-                if (!filtered.length) {
-                    var emptyMessage = !this.autoShown && this.emptyMessage;
-                    if ( typeof emptyMessage == "function")
-                          emptyMessage = this.emptyMessage(prefix);
-                    if (emptyMessage) {
-                        var completionsForEmpty = [{
-                            caption: this.emptyMessage(prefix),
-                            value: ""
-                        }];
-                        this.completions = new FilteredList(completionsForEmpty);
-                        this.openPopup(this.editor, prefix, keepPopupPosition);
-                        return;
+        }).provideCompletions(this.editor, completionOptions,
+            /**
+             * @type {(err: any, completions: FilteredList, finished: boolean) => void | boolean}
+             * @this {Autocomplete & {emptyMessage}}
```

Comment:
shouldn't emptyMessage be on Autocomplete type?

Reply:
yes, you're right, now we could just place it to declare part